File: allure2-androidj/src/main/java/com/yq/allure2_androidj/android/runner/AllureAndroidRunner.java
package com.yq.allure2_androidj.android.runner;

import android.os.Bundle;
import android.support.test.runner.AndroidJUnitRunner;

import com.yq.allure2_androidj.android.listener.AllureAndroidListener;
import com.yq.allure2_androidj.common.Allure;

import java.io.File;


public class AllureAndroidRunner extends AndroidJUnitRunner {

    public void onCreate(Bundle arguments) {
        String listener = "listener";
        CharSequence sequence = arguments.getCharSequence("listener");
        if (sequence != null) {
            sequence = "" + sequence + ',' + AllureAndroidListener.class.getName();
        }else
            sequence = AllureAndroidListener.class.getName();
        arguments.putCharSequence(listener, sequence);

        super.onCreate(arguments);
    }
}

File: allure2-android/src/main/java/com/yq/allure2_android/android/runner/AllureAndroidRunner.kt
package com.yq.allure2_android.android.runner

import android.os.Bundle
import android.support.annotation.Keep
import android.support.test.runner.AndroidJUnitRunner
import com.yq.allure2_android.android.listener.AllureAndroidListener
import com.yq.allure2_android.common.Allure
import com.yq.allure2_android.common.utils.*
import java.io.File

/**
 * @author king
 */

@Suppress("unused")
@Keep
open class AllureAndroidRunner : AndroidJUnitRunner() {

    override fun onCreate(arguments: Bundle) {
        arguments.putCharSequence("listener", arguments.getCharSequence("listener")
                ?.let {
                    "$it,${AllureAndroidListener::class.java.name}"
                }
                ?: AllureAndroidListener::class.java.name)
        super.onCreate(arguments)


//        Allure.resDir = sdDirectory(resultsName)
//
//        deleteFolderFile(Allure.resDir!!, true)
//        mkresultDir()

    }
}